File: claude/mail/scripts/doctor.py
#!/usr/bin/env python3

import json
import os
import sys
from pathlib import Path


store_path = Path(os.environ.get("FROMAIAGENT_PROFILE_STORE_PATH", "") or (Path.home() / ".fromaiagent" / "profiles.json"))
issues = []
warnings = []


def matches_identifier(profile, identifier):
    return bool(identifier) and (profile.get("name") == identifier or profile.get("address") == identifier)


def resolve_default_profile(parsed, profiles):
    if parsed.get("defaultProfile"):
        current = next((profile for profile in profiles if matches_identifier(profile, parsed.get("defaultProfile"))), None)
        if current:
            return current
    if len(profiles) == 1:
        return profiles[0]
    return None

version = sys.version_info
if version < (3, 9):
    issues.append(f"Python {version.major}.{version.minor} detected. Python 3.9+ is recommended.")

if not store_path.exists():
    issues.append(f"Profile store not found: {store_path}")
else:
    try:
        parsed = json.loads(store_path.read_text(encoding="utf-8"))
        profiles = parsed.get("profiles", []) if isinstance(parsed.get("profiles"), list) else []
        if not profiles:
            issues.append("No profiles stored.")
        if len(profiles) > 1 and not parsed.get("defaultProfile"):
            issues.append("Multiple profiles exist but no default profile is selected.")
        if parsed.get("defaultProfile") and not next((profile for profile in profiles if matches_identifier(profile, parsed.get("defaultProfile"))), None):
            issues.append(f"Default profile not found: {parsed.get('defaultProfile')}")
        active = resolve_default_profile(parsed, profiles)
        if active:
            for field in ("publicKey", "privateKey"):
                if not active.get(field):
                    issues.append(f"Active profile is missing {field}.")
            if not active.get("address"):
                warnings.append("Active profile has no address yet. This is normal before create_mailbox without a custom address.")
    except json.JSONDecodeError:
        issues.append(f"Profile store is not valid JSON: {store_path}")

if issues:
    print("Doctor check found issues:")
    for issue in issues:
        print(f"- {issue}")
    raise SystemExit(1)

for warning in warnings:
    print(f"Warning: {warning}")
print("Doctor check passed.")

File: cursor/mail/scripts/doctor.js
#!/usr/bin/env node

const { existsSync, readFileSync } = require("node:fs");
const { homedir } = require("node:os");
const { join } = require("node:path");

const storePath = process.env.FROMAIAGENT_PROFILE_STORE_PATH?.trim() || join(homedir(), ".fromaiagent", "profiles.json");
const major = Number.parseInt(process.versions.node.split(".")[0] || "0", 10);

const issues = [];
const warnings = [];

function matchesIdentifier(profile, identifier) {
  return Boolean(identifier) && (profile.name === identifier || profile.address === identifier);
}

function resolveDefaultProfile(parsed, profiles) {
  if (parsed.defaultProfile) {
    return profiles.find((profile) => matchesIdentifier(profile, parsed.defaultProfile)) || null;
  }
  return profiles.length === 1 ? profiles[0] : null;
}

if (major < 22) {
  issues.push(`Node.js ${process.versions.node} detected. Node 22.6+ is recommended.`);
}

if (!existsSync(storePath)) {
  issues.push(`Profile store not found: ${storePath}`);
} else {
  try {
    const parsed = JSON.parse(readFileSync(storePath, "utf8"));
    const profiles = Array.isArray(parsed.profiles) ? parsed.profiles : [];
    if (profiles.length === 0) {
      issues.push("No profiles stored.");
    }
    if (profiles.length > 1 && !parsed.defaultProfile) {
      issues.push("Multiple profiles exist but no default profile is selected.");
    }
    if (parsed.defaultProfile && !profiles.find((profile) => matchesIdentifier(profile, parsed.defaultProfile))) {
      issues.push(`Default profile not found: ${parsed.defaultProfile}`);
    }
    const active = resolveDefaultProfile(parsed, profiles);
    if (active) {
      for (const field of ["publicKey", "privateKey"]) {
        if (!active[field]) {
          issues.push(`Active profile is missing ${field}.`);
        }
      }
      if (!active.address) {
        warnings.push("Active profile has no address yet. This is normal before create_mailbox without a custom address.");
      }
    }
  } catch {
    issues.push(`Profile store is not valid JSON: ${storePath}`);
  }
}

if (issues.length === 0) {
  for (const warning of warnings) {
    console.log(`Warning: ${warning}`);
  }
  console.log("Doctor check passed.");
} else {
  console.log("Doctor check found issues:");
  for (const issue of issues) {
    console.log(`- ${issue}`);
  }
  process.exitCode = 1;
}
